### DISCUSSION TITLE
Does celery support kafka as broker yet?

### DISCUSSION THREAD

**Discussion by Lunaticf** [celery/celery#7676] [Q&A] [answered]
I can see there is some PR about kafka support merged. Thanks

[a comment by an automated account is not shown]

**Comment by yataosu** [comment, marked as answer]
@Lunaticf From the kombu pr https://github.com/celery/kombu/pull/1506, kombu support kafka in https://github.com/celery/kombu/releases/tag/v5.3.0a1.
Maybe Celery will support kafka in future.

**Comment by byte-kevin** [comment]
Hi @Lunaticf, good news! Kombu supports Kafka, enabling Celery to seamlessly utilize Kafka functionality. I've successfully tested this on my local machine.

Here's my local configuration:

broker_url = "confluentkafka://192.168.16.130:9092/"

To further optimize your setup, consider utilizing the broker_transport_options setting to provide additional configuration to the Kafka client pool. The key features include:

Features:

Type: Virtual
Supports Direct: Yes
Supports Topic: Yes
Supports Fanout: No
Supports Priority: No
Supports TTL: No
Feel free to leverage these features for enhanced performance and flexibility in your Celery setup with Kafka. Let me know if you have any questions or need further assistance!

**Reply by byte-kevin** [reply]
To get started, make sure to add `confluentkafka` to your environment by running:

pip install confluent-kafka

**Reply by auvipy** [reply]
thanks for checking and confirming!

**Reply by glivne** [reply]
Would you subscribe to a topic with /{topic} at the end of the connection string like vhost in mq?

**Reply by glivne** [reply]
Nevermind, I found the solution but I'll leave it here for people who need it. A celery Queue = Kafka Topic. If you want a specific topic, task_default_queue={topic} does the trick. not sure what broker_vhost is used for if at all yet.

**Reply by cuongkane** [reply]
Thank @pengtaochang for your information. 
Have you encountered any more problems so far?

**Reply by Nusnus** [reply]
May I ask you to contribute an enhancement to our docs to match your description? 🙏 @glivne 

Thank you

**Reply by glivne** [reply]
> May I ask you to contribute an enhancement to our docs to match your description? 🙏 @glivne
> 
> Thank you

I wouldn't mind doing that, but I don't have much to add to what I wrote except the following:
"Celery creates a control queue for every worker, so you need permissions to create topics on your Kafka cluster."
At that point the poc idea was dropped because our IT kafka team was not willing to give us a more powerful user, or some segmentation solution for Kafka etc.
I don't have access to this code right now either way so I cannot help you beyond this.
@Nusnus
